### PR TITLE
Feat/vsdspub 471/nifi processors

### PR DESCRIPTION
**Pull request by Yalz**

### CODE REVIEW

File: ldi-core/ldes-client/src/main/java/be/vlaanderen/informatievlaanderen/ldes/client/member/sqlite/SqliteMemberRepository.java [hunk 1]
```diff
@@ -59,6 +60,6 @@ public void addProcessedMember(String memberId) {
 	@Override
 	public void clearState() {
 		removeProcessedMembers();
-		destroyDb();
+    //destroyDb();
```
Comment (Tomvbe) [inline ldi-core/ldes-client/src/main/java/be/vlaanderen/informatievlaanderen/ldes/client/member/sqlite/SqliteMemberRepository.java:63]: why is this commented out?

File: ldi-core/ldes-client/src/main/java/be/vlaanderen/informatievlaanderen/ldes/client/member/sqlite/SqliteMemberRepository.java [hunk 2]
```diff
@@ -31,6 +31,7 @@ public void removeProcessedMembers() {
 		entityManager.getTransaction().commit();
 	}
 
+	@Deprecated(since = "Issue with db still being used")
```
Comment (Tomvbe) [inline ldi-core/ldes-client/src/main/java/be/vlaanderen/informatievlaanderen/ldes/client/member/sqlite/SqliteMemberRepository.java:34]: what exactly does this refer to?

File: ldi-core/sparql-construct/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/SparqlConstructTransformer.java [hunk 3]
```diff
@@ -8,17 +8,17 @@
 
 public class SparqlConstructTransformer implements LdiTransformer {
 	private final Query query;
-	private final boolean inferMode;
```
Comment (Tomvbe) [inline ldi-core/sparql-construct/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/SparqlConstructTransformer.java:11]: I am not sure that we want this as inference is an actual concept within rdf:
https://jena.apache.org/documentation/inference/#overview

File: ldi-nifi/ldi-nifi-processors/sparql-interactions-processor/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/config/SparqlProcessorProperties.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+			.defaultValue("SELECT ?subject ?predicate ?object WHERE {?subject ?predicate ?object}")
+			.addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+			.build();
+
+	public static final AllowableValue REPLACE_ALL = new AllowableValue(
+			"replace",
+			"Replace mode",
+			"In replace mode, the result of the CONSTRUCT query will replace the contents of the flowfile.");
+
+	public static final PropertyDescriptor INCLUDE_ORIGINAL = new PropertyDescriptor.Builder().name("INFERENCE_MODE")
```
Comment (Tomvbe) [inline ldi-nifi/ldi-nifi-processors/sparql-interactions-processor/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/config/SparqlProcessorProperties.java:40]: Either call it inference or original but I would not give it two names..

File: ldi-nifi/ldi-nifi-processors/ldes-client-processor/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/LdesClient.java [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+		return endpointRequester
+				.determineStartingNode(new Endpoint(dataSourceUrl, dataSourceFormat))
+				.map(StartingNode::url)
+				.orElseThrow(
+						() -> new IllegalArgumentException(
+								"Starting url could not be determined for fragmentId: " + dataSourceUrl));
+	}
+
+	@Override
+	public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
```
Comment (Tomvbe) [inline ldi-nifi/ldi-nifi-processors/ldes-client-processor/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/LdesClient.java:96]: we should decrease the complexity of this method

Comment (Yalz) [reply ldi-nifi/ldi-nifi-processors/ldes-client-processor/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/LdesClient.java:96]: Out of scope of this PR, but to be picked up in dedicated dev session

File: ldi-nifi/ldi-nifi-processors/sparql-interactions-processor/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/config/SparqlProcessorProperties.java (hunk 4 above)
Comment (Yalz) [reply ldi-nifi/ldi-nifi-processors/sparql-interactions-processor/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/config/SparqlProcessorProperties.java:40]: Good point! Will fix

File: ldi-core/sparql-construct/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/SparqlConstructTransformer.java (hunk 3 above)
Comment (Yalz) [reply ldi-core/sparql-construct/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/SparqlConstructTransformer.java:11]: Isn't this a name you originally implemented? 😅

**Pull request merged**